llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7403,7 +7404,23 @@ InstructionCost LoopVectorizationPlanner::cost(VPlan &Plan,
 
   // Now compute and add the VPlan-based cost.
   Cost += Plan.cost(VF, CostCtx);
-  LLVM_DEBUG(dbgs() << "Cost for VF " << VF << ": " << Cost << "\n");
+#ifndef NDEBUG
+  unsigned EstimatedWidth = VF.getKnownMinValue();
+  if (VF.isScalable())
+    if (std::optional<unsigned> VScale = getVScaleForTuning(OrigLoop, TTI))
+      EstimatedWidth *= *VScale;
+  LLVM_DEBUG(dbgs() << "Cost for VF " << VF << ": " << Cost
+                    << " (Estimated cost per lane: ");
+  if (Cost.isValid()) {
+    std::string CostString;
+    raw_string_ostream CostStream(CostString);
+    double CostPerLane = double(*Cost.getValue()) / EstimatedWidth;
+    llvm::write_double(CostStream, CostPerLane, FloatStyle::Fixed, 1);
```

Comment:
Could this make use of `dbgs() << format("%1.f", CostPerLane)`?

Reply:
Hmm, I was looking for something like that but couldn't find it. Thanks - I'll give it a try!